extensions/compose/src/extension.ts

```diff
@@ -216,56 +214,68 @@ export async function activate(extensionContext: extensionApi.ExtensionContext):
   // Push the CLI tool as well (but it will do it postActivation so it does not block the activate() function)
   // Post activation
   setTimeout(() => {
-    postActivate(extensionContext, composeDownload).catch((error: unknown) => {
+    postActivate(composeDownload, detect).catch((error: unknown) => {
       console.error('Error activating extension', error);
     });
   }, 0);
 }
 
 // Activate the CLI tool (check version, etc) and register the CLi so it does not block activation.
-async function postActivate(
-  extensionContext: extensionApi.ExtensionContext,
-  composeDownload: ComposeDownload,
-): Promise<void> {
-  // The location of the binary (local storage folder)
-  const binaryPath = path.join(
-    extensionContext.storagePath,
-    'bin',
-    os.isWindows() ? composeCliName + '.exe' : composeCliName,
-  );
-  let binaryVersion = '';
-
-  // Retrieve the version of the binary by running exec with --short
-  try {
-    const result = await extensionApi.process.exec(binaryPath, ['--version', '--short']);
-    binaryVersion = result.stdout;
-  } catch (e) {
-    console.error(`Error getting compose version: ${e}`);
+async function postActivate(composeDownload: ComposeDownload, detect: Detect): Promise<void> {
+  // build executable name for current platform
+  const executable = os.isWindows() ? composeCliName + '.exe' : composeCliName;
+
+  // binary info
+  let binaryInfo: { version: string; path: string; updatable?: boolean } | undefined = undefined;
+
+  // let's check for system-wide
+  const installedSystemWide = await detect.checkSystemWideDockerCompose();
```

Comment:
nit: `Detect.checkSystemWideDockerCompose()` and `detect.getDockerComposeBinaryInfo()` run exactly the same command. There should be just one call of `detect.getDockerComposeBinaryInfo()` used in try/catch block to have the same information about compose system wide availability, but one compose execution less in case it is installed.